File: rb/42.rb
def value word
  word.chars.map { |c| c.ord - 64 }.reduce(:+)
end

# Build array of triangular numbers.
tri = [0]
20.times do |i|
  tri << tri[-1] + i
end

words = IO.read('../input/42').scan(/\w+/)
p words.select { |w| tri.include?(value w) }.size

File: hs/42.hs
import Data.Char (ord)

triangular n = foldl1 (+) [1..n]

triangles n = map triangular [1..n]

value = foldl1 (+) . map (\c -> ord c - 64)

main = do
    input <- readFile "../input/42"
    let words = (read ("[" ++ input ++ "]") :: [String])
    print $ length $ filter (\n -> elem n $ triangles 20) $ map value words
